packages/build/tests/telemetry/tests.js

```diff
@@ -0,0 +1,139 @@
+'use strict'
+
+const test = require('ava')
+const uuid = require('uuid')
+
+const { runFixture } = require('../helpers/main')
+const { startServer } = require('../helpers/server.js')
+
+// Normalize telemetry request so it can be snapshot
+const normalizeSnapshot = function ({ body, ...request }) {
+  return { ...request, body: normalizeBody(body) }
+}
+
+const normalizeBody = function ({
+  timestamp,
+  anonymousId,
+  properties: { duration, buildVersion, nodeVersion, osPlatform, osName, ...properties } = {},
+  ...body
+}) {
+  const optAnonymousId = anonymousId ? { anonymousId: uuid.validate(anonymousId) ? 'uuid' : 'invalid' } : {}
+  const optDuration = duration ? { duration: typeof duration } : {}
+  return {
+    ...body,
+    ...optAnonymousId,
+    timestamp: typeof timestamp,
+    properties: {
+      ...properties,
+      ...optDuration,
+      buildVersion: typeof buildVersion,
+      nodeVersion: typeof nodeVersion,
+      osPlatform: typeof osPlatform,
+      osName: typeof osName,
+    },
+  }
+}
+
+const SITE_INFO_PATH = '/api/v1/sites/test'
+
+const SITE_INFO_DATA = {
+  id: 'test',
+  user_id: 'its-me-test',
+}
+
+const TELEMETRY_PATH = '/track'
+
+const runWithApiMock = async function (
+  t,
+  fixture,
+  { telemetryOrigin, env = {}, snapshot = false, runApi = false, ...flags } = {},
+) {
+  // Start the mock telemetry server
+  const {
+    scheme: schemeTelemetry,
+    host: hostTelemetry,
+    requests: telemetryRequests,
+    stopServer: stopTelemetryServer,
+  } = await startServer({
+    path: TELEMETRY_PATH,
+  })
+  const stopServers = [stopTelemetryServer]
+  const testOpts = { telemetryOrigin: telemetryOrigin || `${schemeTelemetry}://${hostTelemetry}` }
+
+  // Start the Netlify API server mock
+  if (runApi) {
+    const { host, scheme, stopServer: stopAPIServer } = await startServer({
+      path: SITE_INFO_PATH,
+      response: SITE_INFO_DATA,
+    })
+    stopServers.push(stopAPIServer)
+    testOpts.scheme = scheme
+    testOpts.host = host
+  }
+
+  try {
+    await runFixture(t, fixture, {
+      flags: { siteId: 'test', testOpts, ...flags },
+      env,
+      snapshot,
+    })
+  } finally {
+    stopServers.forEach((stopServer) => {
+      stopServer()
```

Comment:
This function is `async`, so it probably should be awaited.

Reply:
You're right, thanks! Fixed in https://github.com/netlify/build/pull/2349/commits/9bac3f489eca8ea810088df9bf0107805896f923